### PR TITLE
IGNITE-23287 Throw CompactedException for getAll metastorage methods

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-23287

### CODE REVIEW

File: modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java [hunk 1]
```diff
@@ -241,7 +241,13 @@ public interface MetaStorageManager extends IgniteComponent {
     HybridTimestamp timestampByRevisionLocally(long revision);
 
     /**
-     * Retrieves entries for given keys.
+     * Returns a future of getting the latest version of an entries by keys from the metastore leader.
```
Comment (sashapolo) [inline modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java:244]:
```suggestion
     * Returns a future containing the latest version of entries corresponding to the given keys from the metastore leader.
```

File: modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java [hunk 2]
```diff
@@ -241,7 +241,13 @@ public interface MetaStorageManager extends IgniteComponent {
...
+     *
+     * <p>Never completes with a {@link CompactedException}.</p>
+     *
+     * <p>Future may complete with {@link NodeStoppingException} if the node is in the process of stopping.</p>
+     *
+     * @param keys Not empty keys.
```
Comment (sashapolo) [inline modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java:250]:
```suggestion
     * @param keys Set of keys (must not be empty).
```

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java [hunk 3]
```diff
@@ -60,8 +60,7 @@ public interface KeyValueStorage extends ManuallyCloseable {
      *
      * <p>Never throws {@link CompactedException}.</p>
      *
-     * @param key The key.
-     * @return Value corresponding to the given key.
```
Comment (sashapolo) [inline modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java:64]: What is this change for?

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java [hunk 4]
```diff
@@ -132,21 +131,23 @@ public interface KeyValueStorage extends ManuallyCloseable {
     List<Entry> get(byte[] key, long revLowerBound, long revUpperBound);
 
     /**
-     * Returns all entries corresponding to given keys.
+     * Returns the latest version of an entries by keys.
```
Comment (sashapolo) [inline modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java:134]:
```suggestion
     * Returns the latest version of entries corresponding to the given keys
```

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java [hunk 5]
```diff
@@ -132,21 +131,23 @@ public interface KeyValueStorage extends ManuallyCloseable {
...
-     * @param keys Keys collection.
-     * @return Entries corresponding to given keys.
+     * @param keys Not empty keys.
      */
-    Collection<Entry> getAll(List<byte[]> keys);
+    List<Entry> getAll(List<byte[]> keys);
 
     /**
-     * Returns all entries corresponding to given keys and bounded by the given revision.
+     * Returns an entries by the given keys and bounded by the given revision.
```
Comment (sashapolo) [inline modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java:143]:
```suggestion
     * Returns entries corresponding to the given keys and bounded by the given revision.
```

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java [hunk 6]
```diff
@@ -132,21 +131,23 @@ public interface KeyValueStorage extends ManuallyCloseable {
...
      *
-     * @param keys Keys collection.
+     * @param keys Not empty keys.
      * @param revUpperBound Upper bound of revision.
-     * @return Entries corresponding to given keys.
+     * @throws CompactedException If getting one of an entries will throw this exception as if using {@link #get(byte[], long)}.
```
Comment (sashapolo) [inline modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java:147]:
```suggestion
     * @throws CompactedException If getting any of the individual entries would have thrown this exception as if {@link #get(byte[], long)} was used.
```

File: modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java (hunk 2 above)
Comment (tkalkirill) [reply modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java:250]: thanks

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java (hunk 5 above)
Comment (tkalkirill) [reply modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java:143]: Thanks.

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java (hunk 4 above)
Comment (tkalkirill) [reply modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java:134]: Thanks.

File: modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java (hunk 1 above)
Comment (tkalkirill) [reply modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java:244]: Thanks.

**Review by sashapolo — APPROVED** [review]

**Pull request merged**